src/Components/test/E2ETest/ServerRenderingTests/ErrorHandlingTest.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using TestServer;
+using Components.TestServer.RazorComponents;
+using Microsoft.AspNetCore.E2ETesting;
+using Xunit.Abstractions;
+using OpenQA.Selenium;
+
+namespace Microsoft.AspNetCore.Components.E2ETests.ServerRenderingTests;
+
+public class ErrorHandlingTest(BrowserFixture browserFixture, BasicTestAppServerSiteFixture<RazorComponentEndpointsStartup<App>> serverFixture, ITestOutputHelper output)
+    : ServerTestBase<BasicTestAppServerSiteFixture<RazorComponentEndpointsStartup<App>>>(browserFixture, serverFixture, output)
+{
+
+    [Fact]
+    public void RendersExceptionFromComponent()
+    {
+        GoTo("Throws");
+
+        Browser.Equal("Error", () => Browser.Title);
+
+        Assert.Collection(
+            Browser.FindElements(By.CssSelector(".text-danger")),
+            item => Assert.Equal("Error.", item.Text),
+            item => Assert.Equal("An error occurred while processing your request.", item.Text));
```

Comment:
Can this test verify be extended that we don't activate interactive components?

I think the difficult situation we wanted to deal with is if the `Router` is interactive, because that would cause an infinite loop of errors.

Reply:
I've added a change to make sure that we don't include any marker for interactive components